src/directives/model/checkbox.js

```diff
@@ -5,17 +5,52 @@ module.exports = {
   bind: function () {
     var self = this
     var el = this.el
+    var trueExp = this._checkParam('true-value')
+    var falseExp = this._checkParam('false-value')
+
+    function getValue () {
+      var val = el.checked
+      if (val && trueExp !== null) {
+        val = self.vm.$eval(trueExp)
+      }
+      if (!val && falseExp !== null) {
+        val = self.vm.$eval(falseExp)
+      }
+      return val
+    }
+    this._getValue = getValue
+
+    function matchValue (value) {
+      var trueValue = true
```

Comment:
Since `el.checked` can be only `true` or `false`, we just need to check if `value` equals `trueValue` here - in all other cases we just return false. This could make `matchValue` much simpler.